### PR TITLE
Add MSstatsTMT

### PR DESCRIPTION
**Pull request by npinter**
MSstatsTMT provides statistical tools for detecting differentially abundant proteins in shotgun mass spectrometry-based proteomic experiments with tandem mass tag (TMT) labeling.

### CODE REVIEW

File: tools/msstatstmt/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+categories:
+  - Proteomics
+description: MSstatsTMT protein significance analysis in shotgun mass spectrometry-based proteomic experiments with tandem mass tag (TMT) labeling
+long_description: |
+    MSstatsTMT provides statistical tools for detecting differentially abundant proteins in shot-gun mass spectrometry-based proteomic experiments with tandem mass tag (TMT) labeling.
+owner: galaxyp
+remote_repository_url: https://github.com/galaxyproteomics/tools-galaxyp/tree/master/tools/msstatstmt
```
Comment (bgruening) [inline tools/msstatstmt/.shed.yml:7]: there is a repo name missing isn't it?

Comment (npinter) [reply tools/msstatstmt/.shed.yml:7]: yes, indeed! I've used the msstats wrapper as a base.. it's missing there too.

File: tools/msstatstmt/msstatstmt.xml [hunk 2]
```diff
@@ -36,20 +44,34 @@ library('MSstatsTMT', warn.conflicts = F, quietly = T, verbose = F)
...
-                                  remove_norm_channel = $proteinSummarization.remove_norm_channel,
-                                  remove_empty_channel = $proteinSummarization.remove_empty_channel,
-                                  MBimpute = $proteinSummarization.MBimpute,
-                                  #if $proteinSummarization.maxQuantileforCensored.select == 'false'
-                                  maxQuantileforCensored = NULL)
-                                  #else
-                                  maxQuantileforCensored = $proteinSummarization.maxQuantileforCensored.value)
-                                  #end if
+#elif $input.input_src == 'OpenMS'
+    #if $input.oms_input.is_of_type('csv')
```
Comment (bgruening) [inline tools/msstatstmt/msstatstmt.xml:48]: If you like I think you can drop the support of CSV. If a user realy needs to use CSV, galaxy can/should convert CSV to tabular before you use the tool.

File: tools/msstatstmt/msstatstmt.xml [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+                    <param name="select" value="true"/>
+                    <param name="comparison_matrix" ftype="tabular" value="comparison_matrix.txt"/>
+                </conditional>
+            </conditional>
+            <param name="selected_outputs" value="ProfilePlot"/>
+            <conditional name="which_Protein">
+                <param name="select" value="list"/>
+                <param name="protein_list" ftype="tabular" value="proteinIDs.txt"/>
+            </conditional>
+            <output name="out_group_comp" file="ComparisonResultMQ.tsv" compare="sim_size"/>
```
Comment (bgruening) [inline tools/msstatstmt/msstatstmt.xml:314]: sim_size for a tsv is not good. What is the problem with those test? Would it be better to use asserts? e.g. how many columns, rows, maybe check if some values are contained?

File: tools/msstatstmt/msstatstmt.xml [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+        <param name="selected_outputs" type="select" multiple="true" optional="false" label="Select Outputs">
+            <option value="msstats_log" selected="true">MSstats log</option>
+            <option value="msstatstmt_log" selected="true">MSstatsTMT log</option>
+            <option value="r_script" selected="false">MSstats Rscript</option>
+            <option value="ProfilePlot" selected="false">Profile Plot</option>
+            <option value="QCPlot" selected="false">QC Plot</option>
+        </param>
+        <section name="out_plots_opt" title="Plot Output Options" expanded="false">
+            <param name="ylimUp" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="false" label="Upper limit for y-axis in the log scale" help="No (Default) for Profile Plot and QC Plot uses the upper limit as rounded off maximum of log2(intensities) after normalization + 3.."/>
+            <param name="ylimDown" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="false" label="Lower limit for y-axis in the log scale" help="No (Default) for Profile Plot and QCPlot uses 0.."/>
```
Comment (foellmelanie) [inline tools/msstatstmt/msstatstmt.xml:224]: there are two full stops (..) in the end of help sentence

File: tools/msstatstmt/msstatstmt.xml [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+                            width = $out_plots_opt.width,
+                            height = $out_plots_opt.height,
+                            #if $out_plots_opt.which_Protein.select != 'list'
+                            which.Protein = "$out_plots_opt.which_Protein.select",
+                            #else
+                            which.Protein = unlist(read.table("$out_plots_opt.which_Protein.protein_list", sep = "\n", header = FALSE), use.names = FALSE),
+                            #end if
+                            originalPlot = $out_plots_opt.originalPlot,
+                            summaryPlot = $out_plots_opt.summaryPlot,
+                            address = "")
```
Comment (foellmelanie) [inline tools/msstatstmt/msstatstmt.xml:91]: address line could be removed

File: tools/msstatstmt/msstatstmt.xml [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+                <param name="remove_norm_channel" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="true" label="Remove ’Norm’ channels from protein level data."/>
+                <param name="remove_empty_channel" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="true" label="Remove ’Empty’ channels from protein level data"/>
+                <param name="MBimpute" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="true" label="MBimpute" help="Only for 'method = msstats'. Yes (default) imputes missing values by accelerated failure time (AFT) model. No uses minimum value to impute the missing value for each peptide precursor ion."/>
+                <conditional name="maxQuantileforCensored">
+                    <param name="select" type="select" label="Maximum quantile for deciding censored missing value">
+                        <option value="false">No</option>
```
Comment (foellmelanie) [inline tools/msstatstmt/msstatstmt.xml:174]: We assume missing values are censored. maxQuantileforCensored is Maximum quantile for deciding censored missing value, for instance, 0.999. Default is Null.
--> I assume this means more an optional float

File: tools/msstatstmt/msstatstmt.xml [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+                </assert_contents>
+            </output>
+            <output name="out_profile_plot" file="ProfilePlot.pdf" compare="sim_size"/>
+            <output name="out_qc_plot" file="QCPlot.pdf" compare="sim_size"/>
+        </test>
+        <test expect_num_outputs="2">
+            <conditional name="input">
+                <param name="input_src" value="MaxQuant"/>
+                <param name="evidence" ftype="tabular" value="evidence.txt"/>
+                <param name="annotation" ftype="tabular" value="annotation.txt"/>
```
Comment (foellmelanie) [inline tools/msstatstmt/msstatstmt.xml:299]: better order would be evidence, proteingroups, annotation (msstats labelfree tool will also be changed to that order)

File: tools/msstatstmt/msstatstmt.xml [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+                <param name="group_comparison" value="true"/>
+            </conditional>
+            <param name="selected_outputs" value="msstats_log"/>
+            <output name="out_group_comp">
+                <assert_contents>
+                    <has_n_lines n="51" />
+                    <has_n_columns n="8" />
+                    <has_text text="Long_LF-Short_HF" />
+                    <has_text text="sp|O35226|PSMD4_MOUSE" />
+                </assert_contents>
```
Comment (foellmelanie) [inline tools/msstatstmt/msstatstmt.xml:340]: maybe add a test for the 10 column file format as well

**Review by foellmelanie — COMMENTED** [review]
Looks great and works well, thank you very much!

File: tools/msstatstmt/msstatstmt.xml (hunk 6 above)
Comment (npinter) [reply tools/msstatstmt/msstatstmt.xml]: Thank you for pointing this out! It's much simpler now..

**Review by foellmelanie — APPROVED** [review]
Thank you for the changes, looks good to me now :)

**Comment by bgruening** [comment]
Awesome work!

**Pull request merged**